src/Compiler/ES/ES.idr

```diff
@@ -248,6 +248,8 @@ jsOp (Cast CharType IntType) [x] = pure $ toBigInt $ x ++ ".codePointAt(0)"
 jsOp (Cast CharType IntegerType) [x] = pure $ toBigInt $ x ++ ".codePointAt(0)"
 jsOp (Cast DoubleType IntType) [x] = boundedInt 63 $ "BigInt(Math.floor(" ++ x ++ "))"
 jsOp (Cast DoubleType IntegerType) [x] = pure $ "BigInt(Math.floor(" ++ x ++ "))"
+jsOp (Cast IntType DoubleType) [x] = pure $ "parseFloat(" ++ x ++ ")"
+jsOp (Cast IntegerType DoubleType) [x] = pure $ "parseFloat(" ++ x ++ ")"
```

Comment:
Shouldn't this be `Number(x)` instead of print to string and parse?

Reply:
The `x` in this function is an _Idris string_ (either the Literal, like "100n", or the name of the JS variable) which represents a _JavaScript BigInt value_. So the actual number is not converted to a string.

Reply:
But why not use Number instead? It would seam less arcane.

Reply:
Yeah that makes sense, I'll update the PR.

Reply:
According to MDN, parseFloat _does_ convert its argument to a String first, which is really unneccessary here, so @stepancheg, sorry I misunderstood your initial concern.